File: p028/ruby/p028.rb
#!/usr/bin/ruby

LIMIT = 5

result, n = 1, 1

(3..LIMIT).step(2) do |i|
  n += (i - 2)
  4.times do 
    n += (i - 2) 
    result += n
    p n
  end
end

p result

File: p028/c/p028.c
#include <stdio.h>

#define LIMIT 1001

int main(){

  int i, j, step = 0;
  long result = 1, n = 1;

  for (i = 3; i <= LIMIT; i += 2) {
    step += 2;
    for (j = 0; j < 4; j++) {
      n += step;
      result += n;
    }
  }

  printf("%ld\n", result);
  return 0;
}
